src/core/common/message.hpp

```diff
@@ -291,6 +291,7 @@ class Message : public otMessage, public Buffer
         kSubTypeMleChildUpdateRequest  = 8,  ///< MLE Child Update Request
         kSubTypeMleDataResponse        = 9,  ///< MLE Data Response
         kSubTypeMleChildIdRequest      = 10, ///< MLE Child ID Request
+        kSubTypeMleDataRequest         = 11, ///< MLE Data Request
```

Comment:
It seems only Delayed MLE Data Request need append timestamp, with this new subtype, [`bool mAppendTimestamps : 1;` ](https://github.com/openthread/openthread/pull/6995/files#diff-d9a5f9e4b09beeb00a14a705a02508debf4f9e52cd4a02fe757ca3b07053f041R1664) in #6995 might not be necessary now?

Reply:
Good idea. I think it should work.

We can still get #6995 merged, but revise the implementation in this PR.